lib/Frontend/Frontend.cpp

```diff
@@ -166,6 +167,38 @@ bool CompilerInstance::setup(const CompilerInvocation &Invok) {
   return setUpInputs();
 }
 
+void CompilerInstance::setUpFileSystem() {
+  auto BaseFS = clang::vfs::getRealFileSystem();
+  auto Overlay = llvm::IntrusiveRefCntPtr<clang::vfs::OverlayFileSystem>(
+                    new clang::vfs::OverlayFileSystem(BaseFS));
```

Comment:
Can overlay options refer to files in previous overlays? It's okay if the answer is no, but it probably deserves a comment.

Reply:
This is something we should ask @vsapsai.  Currently, `clang` always uses the base filesystem to look up overlay files.

Reply:
There is a desire to allow file lookup in previous overlays rdar://problem/39440687 And as for me, chaining file systems should work better when we add VFS that doesn't fall back to underlying file system rdar://problem/39465552 I have a change for Clang that moves VFS in that direction but it's not entirely ready yet: not enough tests, some blatantly copy-pasted iterator code. I'll post a draft so that we can align our changes.

Reply:
And here is the promised Clang change https://reviews.llvm.org/D50032

Reply:
(Even with that as the eventual direction I think it's okay to land this as is; just that we may have to go back and change it to layer on VFSs one by one.)